packages/cashier/src/Components/__tests__/recent-transaction.spec.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import RecentTransaction from '../recent-transaction';
+import { createBrowserHistory } from 'history';
+import { Router } from 'react-router';
+
+jest.mock('Stores/connect', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+describe('<RecentTransaction />', () => {
+    const props = {
+        crypto_transactions: [
+            {
+                address_hash: 'tb1ql7w62elx9ucw4pj5lgw4l028hmuw80sndtntxt',
+                address_url:
+                    'https://www.blockchain.com/btc-testnet/address/tb1ql7w62elx9ucw4pj5lgw4l028hmuw80sndtntxt',
+                amount: 0.01,
+                id: '262',
+                is_valid_to_cancel: 1,
+                status_code: 'LOCKED',
+                status_message:
+                    "We're reviewing your withdrawal request. You may still cancel this transaction if you wish. Once we start processing, you won't be able to cancel.",
+                submit_date: 1644408421,
+                transaction_type: 'withdrawal',
+            },
+        ],
+        currency: 'BTC',
+        onMount: jest.fn(),
+        setIsCryptoTransactionsVisible: jest.fn(),
+    };
+
+    const history = createBrowserHistory();
+
+    const getIconName = container => {
+        const icon = container.querySelector('.cashier-recent-transaction__icon');
+        const icon_path = icon.firstChild.getAttribute('xlink:href');
+        return icon_path.slice(icon_path.indexOf('#') + 1);
+    };
+
+    it('should show proper messages', () => {
+        render(
+            <Router history={history}>
+                <RecentTransaction {...props} />
+            </Router>
+        );
+
+        expect(screen.getByText('Recent transactions')).toBeInTheDocument();
+        expect(screen.getByText('Withdrawal BTC')).toBeInTheDocument();
+        expect(screen.getByText('In review')).toBeInTheDocument();
+        expect(screen.getByText('0.01 BTC on Feb 9, 2022')).toBeInTheDocument();
+        expect(screen.getByText('Address:')).toBeInTheDocument();
+        expect(screen.getByText('tb1q....ntxt')).toBeInTheDocument();
+        expect(screen.getByText('Transaction hash:')).toBeInTheDocument();
+        expect(screen.getByText('Pending')).toBeInTheDocument();
+        expect(screen.getByText('View all')).toBeInTheDocument();
+    });
+
+    it('should trigger onClick callback when the user clicks on "View all" button', () => {
+        render(
+            <Router history={history}>
+                <RecentTransaction {...props} />
+            </Router>
+        );
+
+        const view_all_btn = screen.getByText('View all');
```

Comment:
@farrah-deriv unfortunately this is not a button, this is ButtonLink component (eventually this is <a> tag), i can use getByRole method instead of text and use 'link' role:
```
const view_all_btn_link = screen.getByRole('link', { name: 'View all' });
fireEvent.click(view_all_btn_link);
```
is it will be ok?

Reply:
yes that's ok

Reply:
fixed